tests/suites/test_suite_x509parse.function

```diff
@@ -1472,3 +1472,83 @@ exit:
     ;;
 }
 /* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_X509_CRT_PARSE_C */
+void x509_crt_parse_subjectkeyid(data_t *buf, unsigned int subjectKeyIdLength, int ref_ret)
+{
+    mbedtls_x509_crt crt;
+
+    mbedtls_x509_crt_init(&crt);
+
+    TEST_ASSERT(mbedtls_x509_crt_parse_der(&crt, buf->x, buf->len) == ref_ret);
```

Comment:
Use `TEST_EQUAL` when asserting equality.  `TEST_EQUAL` prints both the expected and actual values, which makes debugging failed tests much easier.  I found this out the hard way when debugging #6659.

```suggestion
    TEST_EQUAL(mbedtls_x509_crt_parse_der(&crt, buf->x, buf->len), ref_ret);
```

(and same for all other uses of `TEST_ASSERT` on an equality)

Reply:
I agree, but I will fix this in follow-up PR with test improvements(https://github.com/Mbed-TLS/mbedtls/pull/7411).